downstairs/src/main.rs

```diff
@@ -1341,6 +1411,122 @@ impl Work {
             None
         }
     }
+
```

Comment:
Add a comment here to say what stage of the process this method is responsible for.

Reply:
added in 9dc7ee2